### PR TITLE
Module names now use "_" instead of "-" when plugin names contain "-"

### PR DESCRIPTION
**Pull request by nicoddemus**
Hi @hackebrot,

I run into this issue when creating [pytest-drop-dup-tests](https://github.com/nicoddemus/pytest-drop-dup-tests). :smile:

### CODE REVIEW

**Comment by hackebrot** [comment]
Hi @nicoddemus! :wave: 

Thank you for submitting a PR :blush:

File: hooks/pre_gen_project.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 logging.basicConfig(level=logging.DEBUG)
 logger = logging.getLogger('pre_gen_project')
 
-PLUGIN_REGEX = r'^(?!pytest)[_a-zA-Z][_a-zA-Z0-9]+$'
+PLUGIN_REGEX = r'^(?!pytest)[-_a-zA-Z][-_a-zA-Z0-9]+$'
```
Comment (hackebrot) [inline hooks/pre_gen_project.py]: I feel we should also have a check for ``module_name`` to be pep8 compliant. :smile:

File: tests/test_create_template.py [hunk 2]
```diff
@@ -26,11 +26,13 @@ def run_tox(plugin):
 
 def test_run_cookiecutter_and_plugin_tests(cookies):
     """Create a new plugin via cookiecutter and run its tests."""
-    result = cookies.bake()
+    result = cookies.bake(extra_context={'plugin_name': 'foo-bar'})
```
Comment (hackebrot) [inline tests/test_create_template.py]: :+1:

File: pytest-{{cookiecutter.plugin_name}}/setup.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def read(fname):
     ],
     entry_points={
         'pytest11': [
-            '{{cookiecutter.plugin_name}} = pytest_{{cookiecutter.plugin_name}}',
+            '{{cookiecutter.plugin_name}} = pytest_{{cookiecutter.module_name}}',
```
Comment (hackebrot) [inline pytest-{{cookiecutter.plugin_name}}/setup.py]: This is correct, right? :no_mouth:

Comment (nicoddemus) [inline pytest-{{cookiecutter.plugin_name}}/setup.py]: Yep, the right side should be the module name.

File: hooks/pre_gen_project.py (hunk 1 above)
Comment (nicoddemus) [inline hooks/pre_gen_project.py]: I see, probably all lower case right?

Comment (hackebrot) [inline hooks/pre_gen_project.py]: All lower with underscores if necessary, see [PEP8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names)

File: pytest-{{cookiecutter.plugin_name}}/setup.py (hunk 3 above)
Comment (hackebrot) [inline pytest-{{cookiecutter.plugin_name}}/setup.py]: Okay, cool :+1:

File: hooks/pre_gen_project.py (hunk 1 above)
Comment (nicoddemus) [inline hooks/pre_gen_project.py]: Done! :smile:

Comment (hackebrot) [inline hooks/pre_gen_project.py]: Do you mind changing this to ``logger.error('  More info: {}'.format(link))``?

Comment (nicoddemus) [inline hooks/pre_gen_project.py]: Sure, no problem! Done.

File: tests/test_create_template.py [hunk 4]
```diff
@@ -26,11 +26,13 @@ def run_tox(plugin):
...
 
     assert result.exit_code == 0
     assert result.exception is None
-    assert result.project.basename == 'pytest-foobar'
+    assert result.project.basename == 'pytest-foo-bar'
     assert result.project.isdir()
+    assert result.project.join('pytest_foo_bar.py').isfile()
+    assert result.project.join('tests', 'test_foo_bar.py').isfile()
```
Comment (hackebrot) [inline tests/test_create_template.py]: :+1:

**Comment by hackebrot** [comment]
Again, many thanks Bruno!

**Comment by nicoddemus** [comment]
No worries!

Thanks again for this project, we should make it more prominent on the [docs](http://pytest.org/latest/writing_plugins.html?highlight=cookiecutter#writing-your-own-plugin), probably the first option in that bullet-point! :smile:

**Pull request merged**